src/main/java/org/folio/innreach/config/props/InnReachRetryPolicy.java

```diff
@@ -0,0 +1,45 @@
+package org.folio.innreach.config.props;
+
+import feign.FeignException;
+import lombok.extern.log4j.Log4j2;
+import org.folio.innreach.external.exception.InnReachConnectionException;
+import org.folio.innreach.external.exception.ServiceSuspendedException;
+import org.folio.innreach.external.exception.SocketTimeOutExceptionWrapper;
+import org.springframework.classify.Classifier;
+import org.springframework.retry.RetryPolicy;
+import org.springframework.retry.policy.ExceptionClassifierRetryPolicy;
+import org.springframework.retry.policy.NeverRetryPolicy;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+
+import java.net.SocketTimeoutException;
+
+@Log4j2
+public class InnReachRetryPolicy extends ExceptionClassifierRetryPolicy {
+
+    public InnReachRetryPolicy(int maxAttempts) {
+      final SimpleRetryPolicy simpleRetryPolicy = new SimpleRetryPolicy();
+      simpleRetryPolicy.setMaxAttempts(maxAttempts);
+
+      this.setExceptionClassifier((Classifier<Throwable, RetryPolicy>) classifiable -> {
+
+        if (classifiable instanceof ServiceSuspendedException) {
+            log.info("classifiable is of ServiceSuspendedException");
+          return new NeverRetryPolicy();
+        }
+        else if (classifiable instanceof FeignException) {
+          log.info("classifiable is of FeignException");
+          return new NeverRetryPolicy();
```

Comment:
Doesn't putting `FeignException` here mean that all non 200 exceptions from the HTTP client will not retry on the Spring level and always bubble up to the Kafka level?

If you look at the nested classes here you'll notice that it's pretty much any non-200 HTTP response that will be an instance of `FeignException`.

Reply:
It won't cause issue and for 404 error spring retry will trigger. 

why ? : 
1. When 404 error occurs first **InnReachFeignErrorDecoder** gets triggered and it will be part of **InnReachException** which will be thrown.
2. Now after that **InnReachRetryPolicy** else-if statement will be executed and if you see there is no Classifiable which is handling **InnReachException** , so s**pring retry Template** will get triggered properly for **404 error**. 
3. 
![image](https://user-images.githubusercontent.com/103935659/226569938-3282e15c-1af6-4388-ae03-0f251d206604.png)